aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -243,10 +296,53 @@ impl<'r, 'l> SessionExt<'r, 'l> {
             }
         }
 
-        Ok((change_set_filtered, resource_group_change_set))
+        Ok((
+            change_set_filtered,
+            resource_group_change_set,
+            updated_timestamp,
+        ))
+    }
+
+    // Seen the global timestamp resource being updated.
+    //   returns Ok(Some(timestamp)) if successfully parsed it
+    //   returns Err(()) if failed to parse or seeing a delete, which will result in the call site
+    //     assuming the timestamp to be None
+    fn parse_updated_timestamp(
+        op: &MoveStorageOp<Vec<u8>>,
+    ) -> Result<Option<CurrentTimeMicroseconds>, ()> {
+        match op {
+            MoveStorageOp::New(bytes) | MoveStorageOp::Modify(bytes) => {
+                match bcs::from_bytes(bytes) {
+                    Ok(current_time) => Ok(Some(current_time)),
+                    Err(err) => {
+                        error!("Failed to deserialize CurrentTimeMicroseconds. {}", err);
+                        Err(())
+                    },
+                }
+            },
+            MoveStorageOp::Delete => {
+                error!("CurrentTimeMicroseconds got deleted.");
+                Err(())
+            },
+        }
+    }
+
+    fn get_current_timestamp(
+        updated_timestamp: Result<Option<CurrentTimeMicroseconds>, ()>,
+        remote: &dyn MoveResolverExt,
+    ) -> Option<CurrentTimeMicroseconds> {
+        match updated_timestamp {
+            Ok(Some(timestamp)) => Some(timestamp),
+            Ok(None) => Some(1010),// CurrentTimeMicroseconds::fetch_config(remote),
```

Comment:
why all the corpus have current_timestamp_usecs: 0 instead of the hard coded 1010 here?

Reply:
the last commit is to debug performance regression (which turned out to be within the margin of error), tests were not updated, see the part after "//" for real code.